ambry-cloud/src/main/java/com.github.ambry.cloud/azure/CosmosChangeFeedFindToken.java

```diff
@@ -0,0 +1,239 @@
+   * @param version token version.
+   */
+  public CosmosChangeFeedFindToken(long bytesRead, String startContinuationToken, String endContinuationToken,
+      int index, int totalItems, String cacheSessionId, short version) {
+    this.version = version;
+    this.bytesRead = bytesRead;
+    this.startContinuationToken = startContinuationToken;
+    this.endContinuationToken = endContinuationToken;
+    this.index = index;
+    this.totalItems = totalItems;
+    this.cacheSessionId = cacheSessionId;
+  }
+
+  /**
+   * Deserialize {@link CosmosChangeFeedFindToken} object from input stream.
+   * @param inputStream {@link DataOutputStream} to deserialize from.
+   * @return {@link CosmosChangeFeedFindToken} object.
+   * @throws IOException
+   */
+  public static CosmosChangeFeedFindToken fromBytes(DataInputStream inputStream) throws IOException {
+    CosmosChangeFeedFindToken cosmosChangeFeedFindToken;
+    DataInputStream stream = new DataInputStream(inputStream);
+    short version = stream.readShort();
+    switch (version) {
+      case VERSION_0:
+        FindTokenType type = FindTokenType.values()[stream.readShort()];
+        if (type != FindTokenType.CloudBased) {
+          throw new IllegalArgumentException(
+              String.format("Invalid token type %s found while deserialization. Expected %s.", type,
+                  FindTokenType.CloudBased));
+        }
+        long bytesRead = stream.readLong();
+        String startContinuationToken = Utils.readIntString(inputStream);
+        String endContinuationToken = Utils.readIntString(inputStream);
+        int index = inputStream.readInt();
+        int totalItems = inputStream.readInt();
+        String cacheSessionId = Utils.readIntString(inputStream);
+        cosmosChangeFeedFindToken =
+            new CosmosChangeFeedFindToken(bytesRead, startContinuationToken, endContinuationToken, index, totalItems,
+                cacheSessionId, version);
+        break;
+      default:
+        throw new IllegalStateException("Unknown version of cloud token: " + version);
+    }
+    return cosmosChangeFeedFindToken;
+  }
+
+  /**
+   * Serialize {@link CosmosChangeFeedFindToken} to byte array.
+   * @return serialized byte array.
+   */
+  public byte[] toBytes() {
+    byte[] buf = new byte[size()];
+    ByteBuffer bufWrap = ByteBuffer.wrap(buf);
+    bufWrap.putShort(getVersion());
+    bufWrap.putShort((short) getType().ordinal());
+    bufWrap.putLong(getBytesRead());
+    Utils.serializeNullableString(bufWrap, startContinuationToken);
+    Utils.serializeNullableString(bufWrap, endContinuationToken);
+    bufWrap.putInt(index);
+    bufWrap.putInt(totalItems);
+    Utils.serializeNullableString(bufWrap, cacheSessionId);
+    return buf;
+  }
+
+  /**
+   * Calculate size of the token.
+   * @return size of the token.
+   */
+  public int size() {
+    return 2 * Short.BYTES + Long.BYTES + 5 * Integer.BYTES + getNullableStringLength(startContinuationToken)
```

Comment:
Minor: inconsistency where in some places you say Utils.method1() and others you say method2() making use of static import.  Please be consistent.